eternalcore-core/src/main/java/com/eternalcode/core/feature/msg/toggle/MsgToggleServiceImpl.java

```diff
@@ -1,41 +1,42 @@
 package com.eternalcode.core.feature.msg.toggle;
 
+import com.eternalcode.core.feature.msg.MsgPlaceholderSetup;
 import com.eternalcode.core.injector.annotations.Inject;
 import com.eternalcode.core.injector.annotations.component.Service;
+import com.eternalcode.core.placeholder.cache.AsyncPlaceholderCacheRegistry;
+import com.eternalcode.core.placeholder.cache.AsyncPlaceholderCached;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
 
 @Service
 class MsgToggleServiceImpl implements MsgToggleService {
 
     private final MsgToggleRepository msgToggleRepository;
-    private final ConcurrentHashMap<UUID, MsgState> cachedToggleStates;
+    private final AsyncPlaceholderCacheRegistry cacheRegistry;
```

Comment:
słabe rozwiązanie, najlepiej jakby tego cache nie było w repo + nie było tutaj nic placeholderach, bo to tworzy dziwne zależności. Raczej placeholdery widzą API serisów i tyle. Serwisy nie powinny widzieć placeholders